src/context/MessagingContext.js

```diff
@@ -34,26 +34,30 @@ export const MessagingProvider = (props) => {
     let newMessages = rawMessages
       .sort((a, b) => (a.composed_at < b.composed_at ? -1 : 1))
       .reduce((grouped, message) => {
-        if (
-          message.subject === 'Message' ||
-          message.subject === 'Announce Message'
-        ) {
-          let key =
-            message.to[0].recipient !== user.userName
-              ? message[`to`][0].recipient
-              : message['from'].author;
+        if (message.type === 'message') {
+          let key = message.to !== user.userName ? message.to : message.from;
           if (key) {
             if (!grouped[key] && !messages[key]) {
               grouped[key] = [];
             }
             grouped[key].push(message);
           }
-        } else if (message.subject === 'Survey') {
+        } else if (message.type === 'survey') {
           let key = message.survey.id;
           if (!grouped[key]) {
             grouped[key] = [];
           }
           grouped[key].push(message);
+        } else if (message.type === 'announce') {
+          // assume it's an announcement
+          let key = message.subject;
+          if (grouped[key]) {
```

Comment:
One issue with grouping this way is that multiple messages with the same announcement title will get grouped together.  In fact, an announce message is always a group with just one single message - the announcement that gets sent.  All responses are in the individual user channels.  So we don't need to group when we have an announce message - it's a group of 1.

Reply:
I handled that by adding the date to the title if there was already a match